### PR TITLE
[Android] Use Java 8 language features.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
* Enable [Java 8 support](https://developer.android.com/studio/write/java8-support).
* Make use of Java 8 language features such as lambda expressions and method references.

### CODE REVIEW

**Comment by virxkane** [comment]
And what does it give?
One more note, the onyx sdk did not need to be touched at all.

**Comment by Isira-Seneviratne** [comment]
> And what does it give?

It makes the code much simpler.

> One more note, the onyx sdk did not need to be touched at all.

Okay, thanks for the info.

File: android/src/org/coolreader/crengine/Utils.java [hunk 1]
```diff
@@ -89,32 +89,17 @@ private static boolean moveFile(File oldPlace, File newPlace, boolean removeOld)
 			Log.i("cr3", "File " + oldPlace.getAbsolutePath() + " does not exist!");
 			return false;
 		}
-		FileOutputStream os = null;
-		FileInputStream is = null;
-		try {
+		try (FileOutputStream os = new FileOutputStream(newPlace);
```
Comment (virxkane) [inline android/src/org/coolreader/crengine/Utils.java:92]: @Isira-Seneviratne You have changed the order of function calls.
Before:
```java
if (!newPlace.createNewFile())
	return false; // cannot create file
os = new FileOutputStream(newPlace);
is = new FileInputStream(oldPlace);
```
After:
```java
FileOutputStream os = new FileOutputStream(newPlace);
FileInputStream is = new FileInputStream(oldPlace);
if (!newPlace.createNewFile())
	return false;
```
So, now this function will always return false.
https://developer.android.com/reference/java/io/FileOutputStream#FileOutputStream(java.io.File)
https://developer.android.com/reference/java/io/File#createNewFile()

Comment (Isira-Seneviratne) [reply android/src/org/coolreader/crengine/Utils.java:92]: Okay, I'll fix it.

**Pull request merged**